src/utils/tty.py

```diff
@@ -59,6 +59,49 @@ def get_table_string(df, transpose=False, decimal=2):
     )
 
 
+def convert_time_columns(df, time_unit):
+    """
+    Convert time column values based on the specified time unit.
+    Uses the Unit column to identify which columns contain time data.
+    """
+    if time_unit not in config.TIME_UNITS or "Unit" not in df.columns:
+        return df
+
+    # Avoid modifying the original
+    df_copy = df.copy()
+
+    time_rows = df_copy["Unit"].str.lower().str.contains("ns", na=False)
+
+    time_value_columns = ["Avg", "Min", "Max"]
+
+    for col in time_value_columns:
+        if col in df_copy.columns:
+            mask = time_rows
+            if mask.any():
+                try:
+                    numeric_values = pd.to_numeric(
+                        df_copy.loc[mask, col], errors="coerce"
+                    )
+                    df_copy.loc[mask, col] = numeric_values / config.TIME_UNITS[time_unit]
+                except:
+                    pass
+
+    # Update the Unit column
+    if time_rows.any():
+        df_copy.loc[time_rows, "Unit"] = time_unit
+
+    return df_copy
+
+
+def has_time_data(df):
+    """
+    Check if the dataframe contains time data by looking at the Unit column.
+    """
+    if "Unit" not in df.columns:
+        return False
+    return df["Unit"].str.lower().str.contains("ns", na=False).any()
```

Comment:
We might want to add Note for it, or define as a Big global var somewhere to remind us or whoever as editor:
"ns" stands for Nanosec.

lower().str.contains("ns", na=False): if user has accident use NS as something else...:)  not a big deal.